fdbclient/include/fdbclient/MetaclusterManagement.actor.h

```diff
@@ -1647,9 +1726,249 @@ Future<Void> configureTenant(Reference<DB> db,
+		// and have each other as pairs
+		if (newTenantEntry.present()) {
+			if (newTenantEntry.get().tenantState == TenantState::RENAMING_TO &&
+			    oldTenantEntry.get().tenantState == TenantState::RENAMING_FROM &&
+			    newTenantEntry.get().renamePair.present() && newTenantEntry.get().renamePair.get() == self->oldName &&
+			    oldTenantEntry.get().renamePair.present() && oldTenantEntry.get().renamePair.get() == self->newName) {
+				wait(self->ctx.setCluster(tr, oldTenantEntry.get().assignedCluster.get()));
+				self->tenantId = newTenantEntry.get().id;
+				self->configurationSequenceNum = newTenantEntry.get().configurationSequenceNum;
+				CODE_PROBE(true, "Metacluster rename retry in progress");
+				return Void();
+			} else {
+				CODE_PROBE(true, "Metacluster rename new name already exists");
+				throw tenant_already_exists();
+			};
+		} else {
+			if (self->tenantId == -1) {
+				self->tenantId = oldTenantEntry.get().id;
+			}
+			++oldTenantEntry.get().configurationSequenceNum;
+			self->configurationSequenceNum = oldTenantEntry.get().configurationSequenceNum;
+			wait(self->ctx.setCluster(tr, oldTenantEntry.get().assignedCluster.get()));
+			if (oldTenantEntry.get().tenantState != TenantState::READY) {
+				CODE_PROBE(true, "Metacluster unable to proceed with rename operation");
+				throw invalid_tenant_state();
+			}
+		}
+
+		TenantMapEntry updatedOldEntry = oldTenantEntry.get();
+		TenantMapEntry updatedNewEntry(updatedOldEntry);
+		ASSERT(updatedNewEntry.id == updatedOldEntry.id);
+		ASSERT(updatedNewEntry.id == self->tenantId);
+		ASSERT(updatedOldEntry.configurationSequenceNum == self->configurationSequenceNum);
+		ASSERT(updatedNewEntry.configurationSequenceNum == self->configurationSequenceNum);
+		updatedOldEntry.tenantState = TenantState::RENAMING_FROM;
+		updatedNewEntry.tenantState = TenantState::RENAMING_TO;
+		updatedOldEntry.renamePair = self->newName;
+		updatedNewEntry.renamePair = self->oldName;
+
+		ManagementClusterMetadata::tenantMetadata().tenantMap.set(tr, self->oldName, updatedOldEntry);
+		ManagementClusterMetadata::tenantMetadata().tenantMap.set(tr, self->newName, updatedNewEntry);
+
+		// Updated indexes to include the new tenant
+		ManagementClusterMetadata::clusterTenantIndex.insert(
+		    tr, Tuple::makeTuple(updatedNewEntry.assignedCluster.get(), self->newName));
+
+		// Add new name to tenant group. It should already exist since the old name was part of it.
+		managementClusterAddTenantToGroup(
+		    tr, self->newName, updatedNewEntry, &self->ctx.dataClusterMetadata.get(), true);
+		return Void();
+	}
+
+	ACTOR static Future<Void> updateDataCluster(RenameTenantImpl* self, Reference<typename DB::TransactionT> tr) {
+		state Optional<TenantMapEntry> tenantEntry = wait(TenantAPI::tryGetTenantTransaction(tr, self->oldName));
+
+		if (!tenantEntry.present() || tenantEntry.get().id != self->tenantId ||
+		    tenantEntry.get().configurationSequenceNum >= self->configurationSequenceNum) {
```

Comment:
I don't think we need these checks here. `renameTenantTransaction` already does them and we catch the errors from it below. The one exception is the `configurationSequenceNum` check, but arguably this could be happening in `renameTenantTransaction` as well rather than relying on this check being performed as a precondition.